config_template.cfg

```diff
@@ -587,6 +587,9 @@ MARKER_MONITORING = ( airfoil )
 % Format: ( marker name, wall function type, ... )
 MARKER_WALL_FUNCTIONS= ( airfoil, NO_WALL_FUNCTION )
 %
+% Marker(s) of the surface where custom thermal BC's are defined.
+MARKER_CHT = ( NONE )
```

Comment:
The new Python CHT capability is very interesting.

Just too make sure we avoid conflicts: @talbring @oleburghardt, do you see any potential conflicts with the options here for doing CHT through the Python wrapper as opposed to the native implementation you are working on?

Reply:
Concerning the options: No, not really.

We can go for the boundary marker 'MARKER_CHT' (ours is called 'MARKER_CHT_INTERFACE' but that's just a renaming - including all the internal config variables of course - ... I can adapt ours, no trouble whatsoever).

The differences in both implementations could be more problematic.
Both will have some things in common on the solver's level (above, it's either controlled by the python calls or by a special native CHT driver).
I just had a very brief look into the pyWrapper-branch. If I got it correctly, it's "just" that custom boundary values are read in BC_Isothermal_Wall or BC_HeatFlux_Wall when a boundary is (additionally?) marked as CHT.
I would recommend an own BC routine for that. Main reason is that things might be more difficult than either setting one or the other. And additionally, we could set it up in a way that it fits both needs without blowing up the standard solver routines too much.

One other difference is that in this branch the interface values are stored within the geometry class (and not as solver member variables in my case). That's a legitimate way to go, too, I guess. I could imagine to adapt our code here as well so that it makes the boundary condition implementation more generic :-).

Regards, Ole

Reply:
Thank you for the comments @oleburghardt  !

You are right, when the BC_HeatFlux or the BC_Isothermal are *additionally* marked as CHT ( see MARKER_CHT in the config file) they will just read the custom values stored in the geometry points instead of the one specified in the config.

My implementation is very light and does not need to stay the way it is. It is much easier to me to change the implementation than you changing yours. Originally, I am working on the Py wrapper and not on CHT in particular. So if there is a full CHT implementation the wrapper should not interfere with it.

I will have a look to your branch and see how you proceed. However some hints from you might be a valuable help in order to identify the relevant routines:

- How/where do you store the temperature/heat fluxes used for the fluid part of the computation ?
- When/where do you propagate the (fluid) interface temperatures/heat fluxes from the finest to the coarser grids (multi-grid) ?
- You leave the BC_HeatFlux_Wall and BC_Isothermal_Wall as they are and you use a dedicated BC type for CHT, is that right ?

Thanks,

David

Reply:
(Sorry for the late reply - I've been occupied with all these christmas duties... :D)

From a very practical point of view, your branch will be merged into develop first. And from this point on it's me who should adapt (possibly dublicate) variables and especially config options since the user should not be bothered with it. It's not so much stuff anyway, so I'm pretty sure that there will be no interference from your branches side :-).

1. So far, I am only using a heat equation solver that I wrote myself which is - in the fluid domain - weakly coupled to the flow solver. Within the heat solver, the CHT variables are simply stored as class variables. (I am planning to also adding a coupling to the compressible solver. I guess I will do the same thing there, too.) I like the idea to store them in the geometry class.

2. I have not used multigrid technique so far. I don't even know how this is done for fluid-fluid-couplings (does someone?). In either case, the routines in transfer_structure.cpp only see the different solvers, so they shouldn't be bothered if they are instantiated on the finest or a multigrid level I guess... Hm.

3. Yes. There seem to be some primitive methods for CHT which work quite stable but slow, but also slightly more complicated ones. Too much code in the end for putting it into the standard ones, I assume. This way you can also add the BC routine nicely into integration_structure.cpp (case CHT_WALL: ...).  

Regards, Ole